.drone.yml

```diff
@@ -3,31 +3,91 @@ kind: pipeline
 name: pipeline1
 
 steps:
+- name: restore-cache
+  image: plugins/s3-cache
+  settings:
+    pull: true
+    access_key:
+      from_secret: drone_aws_access_key_id
+    secret_key:
+      from_secret: drone_aws_secret_access_key
+    restore: true
+    root: TODO-BUCKET-NAME
+    filename: pipeline1.tar
+
 - name: checkstyle-and-sevntu
   image: maven:3.6.3-adoptopenjdk-8
+  environment:
+    MAVEN_OPTS: "-Dmaven.repo.local=.m2"
   commands:
   - ./.ci/travis/travis.sh checkstyle-and-sevntu
 
 - name: spotbugs-and-pmd
   image: maven:3.6.3-adoptopenjdk-8
+  environment:
+    MAVEN_OPTS: "-Dmaven.repo.local=.m2"
   commands:
   - ./.ci/travis/travis.sh spotbugs-and-pmd
 
+- name: rebuild-cache
+  image: plugins/s3-cache
+  settings:
+    pull: true
+    access_key:
+      from_secret: drone_aws_access_key_id
+    secret_key:
+      from_secret: drone_aws_secret_access_key
+    rebuild: true
+    root: TODO-BUCKET-NAME
+    filename: pipeline1.tar
+    mount:
+    - .m2
```

Comment:
After testing on this branch and proving plugin to be effective, I suggest to add:
```yml
when:
  event: push
  branch:
  - master
```
So uploading of `.m2` cache will be done only on master branch and restoring on all build. This likely will provide better cache hit and also significantly reduce S3 bill.

Reply:
There is an other common way - update cache only if there are changes in pom.xml in master, like we do in github actions.
https://github.com/checkstyle/checkstyle/blob/9a77689fe9a81a4c1ad0c8f4a923b2b40f1fb360/.github/workflows/diff_report.yml#L146

Reply:
I don't see how we can extends S3 plugin for this, alternatives:
1. Build own docker image instead of `plugin/s3-cache` with bash support and add following section to the `.drone.yml`
  ```yml
  commands:
   - export PLUGIN_FILENAME=checkstyle-maven-cache-$$( md5sum pom.xml | head -c 32 )
   - /bin/drone-s3-cache 
  ```
2. Use own scripts instead of plugin

Please let me know your preference

Reply:
No need for extension of plugin, too much mess to support it.

We can have  few zip with artifacts, not a big problem for s3.